### PR TITLE
fix: Add and apply NumberBox CommonStates

### LINKED ISSUE #5543: CommonStates not applied on `NumberBox`
## Current behavior

Properties modified by CommonStates inside NumberBox style are not applied. This issue might be happening on other winui controls.

## Expected behavior

Properties modified by CommonStates inside NumberBox style should be applied. (UWP is working fine/as expected)

## How to reproduce it (as minimally and precisely as possible)

LINE 26 of MainPage.xaml
[NumberBoxState.zip](https://github.com/unoplatform/uno/files/6184005/NumberBoxState.zip)

## Workaround

use states inside each (RepeatButton and TextBox)

## Environment

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):
Uno.UI : 3.7.0-dev.2

Affected platform(s):

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: 16.9.2)
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

### LINKED ISSUE #6401: Get latest NumberBox from MUX
## What would you like to be added:
Update NumberBox to latest v1 code

## Why is this needed:
Alignment

## For which Platform:
All

## Anything else we need to know?
FYI @MartinZikmund

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #5543, closes #6401

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

CommonStates don't exist and hence not applied.

## What is the new behavior?

Added CommonStates and applied them (port from https://github.com/microsoft/microsoft-ui-xaml/pull/3005)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/NumberBox/NumberBoxTests.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+		//		numberBox.InputScope = scope;
+		//		Content.UpdateLayout();
+
+		//		Assert.AreEqual(1, inputTextBox.InputScope.Names.Count);
+		//		Assert.AreEqual(InputScopeNameValue.CurrencyAmountAndSymbol, inputTextBox.InputScope.Names[0].NameValue, "The InputScope should be 'CurrencyAmountAndSymbol'.");
+		//	});
+
+		//	return;
+		//}
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/NumberBox/NumberBoxTests.cs:67]: We don't have `InputScope` or `TextAlignment` properties.

I even can't find them in the API reference [here](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.numberbox?view=winui-3.0), but they're implemented in MUX repo.

https://github.com/microsoft/microsoft-ui-xaml/pull/3186
https://github.com/microsoft/microsoft-ui-xaml/pull/2901

I have no idea if we should port these two PRs and enable those commented tests.

Comment (jeromelaban) [reply src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/NumberBox/NumberBoxTests.cs:67]: We should be updating the numberbox code to the latest, which is likely to include what's missing. It looks like the current implementation is using commit 2237b09d9ba6052414ddccf3d61316de20564303 from winui.

Comment (Youssef1313) [reply src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/NumberBox/NumberBoxTests.cs]: @jeromelaban Attempted to do that in 2a95376. I'm hoping I did it correctly :)

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs [hunk 2]
```diff
@@ -74,8 +72,65 @@ public NumberBox()
...
+		// This was largely copied from Calculator's GetRegionalSettingsAwareDecimalFormatter()
+		private DecimalFormatter GetRegionalSettingsAwareDecimalFormatter()
+		{
+			return new DecimalFormatter
+			{
+				IntegerDigits = 1,
+				FractionDigits = 0,
+			};
+
+			// UNO TODO:
```
Comment (jeromelaban) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs:99]:
```suggestion
			// UNO TODO: https://github.com/unoplatform/uno/issues/6908
```

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
Thanks for your contribution!

**Review by Youssef1313 — COMMENTED** [review]
Important: [Uno.UI.FluentTheme/Resources/PriorityDefault/NumberBox.xaml](https://github.com/unoplatform/uno/blob/master/src/Uno.UI.FluentTheme/Resources/PriorityDefault/NumberBox.xaml) isn't yet updated. I think it should before merging.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs [hunk 3]
```diff
@@ -178,8 +180,22 @@ private void InitializeTemplate()
...
-				textBox.KeyDown += OnNumberBoxKeyDown;
-				registrations.Add(() => textBox.KeyDown -= OnNumberBoxKeyDown);
+				if (SharedHelpers.IsRS3OrHigher())
+				{
+					// Listen to PreviewKeyDown because textbox eats the down arrow key in some circumstances.
+					// UNO Docs: PreviewKeyDown is not implemented. Use KeyDown.
+					//textBox.PreviewKeyDown += OnNumberBoxKeyDown;
+					//registrations.Add(() => textBox.PreviewKeyDown -= OnNumberBoxKeyDown);
+					textBox.KeyDown += OnNumberBoxKeyDown;
+					registrations.Add(() => textBox.KeyDown -= OnNumberBoxKeyDown);
```
Comment (Youssef1313) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs:190]: Not sure this is okay or the correct way to do it.

Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs:190]: This is is the same block as below, let's put this block around `#if !HAS_UNO` for now

Comment (Youssef1313) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs]: @jeromelaban Does the change in 11d2266 match what you meant?

Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs]: That's correct! Thanks!

**Pull request merged**